libs/server/ACL/CommandPermissionSet.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Garnet.server.ACL
+{
+    /// <summary>
+    /// Wraps up command permissions behind a reference so it can be atomically swapped.
+    /// </summary>
+    public sealed class CommandPermissionSet
+    {
+        // Do not move these, initialization order is important
+        private static readonly ushort CommandListLength = GetCommandListLength();
+
+        public static readonly CommandPermissionSet All = new("+@all");
+        public static readonly CommandPermissionSet None = new("");
+
+        // Each bit corresponds to RespCommand + subcommand
+        private readonly ulong[] _commandList;
+
+        private CommandPermissionSet(string description)
+            : this(new ulong[CommandListLength], description)
+        {
+        }
+
+        private CommandPermissionSet(ulong[] commandList, string description)
+        {
+            this._commandList = commandList;
+            this.Description = description;
+
+            // Certain commands can always be run, so initialize with them
```

Comment:
Probably revert this since cmd.IsNoAuth needs to happen at CheckACL site anyway ...